ruby_event_store/lib/ruby_event_store/specification.rb

```diff
@@ -4,68 +4,82 @@ module RubyEventStore
   class Specification
     DEFAULT_BATCH_SIZE = 100
 
-    class Result < Struct.new(:direction, :start, :count, :stream, :read_as, :batch_size)
-      def limit?
-        !count.nil?
-      end
+    attr_reader :start, :batch_size
+    def limit?
+      !@count.nil?
+    end
 
-      def limit
-        count || Float::INFINITY
-      end
+    def count
+      @count || Float::INFINITY
+    end
 
-      def global_stream?
-        stream.global?
-      end
+    def global_stream?
+      @stream.global?
+    end
 
-      def stream_name
-        stream.name
-      end
+    def stream_name
+      @stream.name
+    end
 
-      def head?
-        start.equal?(:head)
-      end
+    def head?
+      start.equal?(:head)
+    end
 
-      def forward?
-        direction.equal?(:forward)
-      end
+    def forward?
+      @direction.equal?(:forward)
+    end
 
-      def backward?
-        !forward?
-      end
+    def backward?
+      @direction.equal?(:backward)
+    end
 
-      def batched?
-        read_as.equal?(:batch)
-      end
+    def all?
+      @read_as.equal?(:all)
+    end
 
-      def first?
-        read_as.equal?(:first)
-      end
+    def batched?
+      @read_as.equal?(:batch)
+    end
 
-      def last?
-        read_as.equal?(:last)
-      end
+    def first?
+      @read_as.equal?(:first)
     end
-    private_constant :Result
 
-    # @api private
-    # @private
-    attr_reader :result
+    def last?
+      @read_as.equal?(:last)
+    end
 
-    # @api private
-    # @private
-    def initialize(repository, mapper, result = Result.new(:forward, :head, nil, Stream.new(GLOBAL_STREAM), :all, DEFAULT_BATCH_SIZE))
+    def initialize(repository, mapper,
```

Comment:
Why did you remove the `private` marker? We don't want to document this method, it is not intended to be used publicly by RES users.

Hmmm. It seems this we dropped `Result` class and now this is a `Specification` initializer. Would be nice to move it to the top.

Reply:
I would consider if `Specification` is a private class, not intended for public documentation.